client/src/components/AppCommandBar/AppCommandBar.test.js

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import { render, unmountComponentAtNode } from "react-dom";
+import { act, Simulate } from "react-dom/test-utils";
+import { findByText, findByLabelText, waitFor, findByTestId, fireEvent, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import PubSub from "pubsub-js";
+import AppCommandBar from "./AppCommandBar";
+
+import { configService } from "../../services/ConfigService";
+import { apiService } from "../../services/ApiService";
+import { eventService } from "../../services/EventService";
+
+import initIcons from "../../services/IconService/IconService";
+
+initIcons();
+
+jest.mock("../../services/ConfigService");
+jest.mock("../../services/ApiService");
+jest.mock("../../services/EventService");
+
+const getAllTwins = jest.spyOn(apiService, "getAllTwins");
+const setConfig = jest.spyOn(configService, "setConfig");
+
+
+
+const mockSuccesResponse = { "Status": "Success"};
+
+const optionalComponentsState = [{
+    id: "console",
+    name: "Console",
+    show: false,
+    showProp: "showConsole"
+  },{
+    id: "output",
+    name: "Output",
+    show: false,
+    showProp: "showOutput"
+  }
+];
+
+let toggleCounter = 0;
+
+const toggleOptionalComponent = () => {
+  toggleCounter++;
+};
```

Comment:
Let's not use a global variable for multiple test cases. Best to define these inside the cases.